channeldb/channel_test.go

```diff
@@ -1530,3 +1533,77 @@ func TestFinalHtlcs(t *testing.T) {
 	_, err = cdb.LookupFinalHtlc(chanID, unknownHtlcID)
 	require.ErrorIs(t, err, ErrHtlcUnknown)
 }
+
+// TestSerializeHTlCs tests serialization of HTLCs combined with extra data.
+func TestSerializeHTlCs(t *testing.T) {
+	t.Parallel()
+
+	mockHtlc := HTLC{
+		Signature:     testSig.Serialize(),
+		Incoming:      false,
+		Amt:           10,
+		RHash:         key,
+		RefundTimeout: 1,
+		OnionBlob:     mockOnion(),
+		ExtraData:     make([]byte, 0),
+	}
+
+	testCases := []struct {
+		name  string
+		htlcs []HTLC
+	}{
+		{
+			// Serialize multiple HLTCs with no extra data to
+			// assert that there is no regresion for HTLCs with
+			// no extra data.
+			name: "no extra data",
+			htlcs: []HTLC{
+				mockHtlc, mockHtlc,
+			},
+		},
+		{
+			name: "mixed extra data",
```

Comment:
Could also check a large extra data(over 66000) can cause an error.

Reply:
I don't think that this is the right place for that type of coverage? `ReadVarBytes` max length coverage should be covered [downstream](https://github.com/btcsuite/btcd/blob/9c16d23918b15c468c5647c388b9b7db3bc48dc7/wire/common_test.go#L683).